src/main/java/Polygon.java

```diff
@@ -0,0 +1,9 @@
+public abstract sealed class Polygon implements Shape permits Section, Triangle {
+    private int corners;
+
+    public Polygon(int corners) {
```

Comment:
tu mam wątpliwości czy to jest dobre podejście - zamiast liczby wierzchołków to raczej należałoby przekazywać ich listę

Reply:
racja, dzięki za tę uwagę - poprawiam